installers/install.sh

```diff
@@ -58,14 +57,41 @@ run() {
         native_loc="https://github.com/tridactyl/native_messenger/releases/download/$native_version/native_main-$binary_suffix"
     fi
 
-    mkdir -p "$manifest_home" "$XDG_DATA_HOME"
+    install_to "$manifest_home" "$manifest_home"
 
-    manifest_file="$manifest_home/tridactyl.json"
-    native_file="$XDG_DATA_HOME/native_main"
+    for flatpak_dir in ~/.var/app/*/.mozilla; do
+        [ -d "$flatpak_dir" ] || continue
+        echo
+        echo "Detected flatpak installation in $flatpak_dir"
+        install_to "$flatpak_dir/native-messaging-hosts/" "$HOME/.mozilla/native-messaging-hosts/"
+    done
 
-    echo "Installing manifest here: $manifest_home"
-    echo "Installing script here: XDG_DATA_HOME: $XDG_DATA_HOME"
+    echo
+    echo "Successfully installed Tridactyl native messenger!"
+    echo "Run ':native' in Firefox to check."
+}
+
+# install_to takes two arguments:
+#  1. The path to the manifest home as seen by the install script.
+#  2. The path to the manifest home as seen by Firefox.
+#
+# For regular Firefox installations, these are the same.  They are different in
+# the case of sandboxed installations (like Flatpak).
+install_to() {
+    manifest_home_on_host="$1"
+    manifest_home_in_sandbox="$2"
+
+    mkdir -p "$manifest_home_on_host"
 
+    manifest_file="$manifest_home_on_host/tridactyl.json"
+    # For Flatpak installations, we must install the native binary inside
+    # `~/.mozilla/native-messaging-hosts` as well, because everything outside
+    # of `~/.mozilla` is wiped when restarted.
+    native_binary_name="tridactyl_native_main"
+    native_file="$manifest_home_on_host/$native_binary_name"
```

Comment:
I don't think we're supposed to put anything except the JSON manifests in this folder but it probably doesn't matter.

Is there a good reason for abandoning the old `$XDG_DATA_HOME/.local/share` directory on non-flatpak installations?

I can guarantee that some people will have ill-advised scripts that are rely on the location of the executable, e.g. `:! ls ../../..` - I would prefer it if we could refrain from breaking them